config/query_test.go

```diff
@@ -0,0 +1,113 @@
+
+		// Aggregation
+		{"Aggregation with GROUP BY and allowed tables", "SELECT category_id, COUNT(*) as item_count, AVG(value) as average_value FROM config_items GROUP BY category_id", []string{"config_"}, true},
+		{"Aggregation with GROUP BY and allowed tables", "SELECT category_id, COUNT(*) as item_count, AVG(value) as average_value FROM config_items GROUP BY category_id", []string{"components_"}, false},
+
+		// HAVING
+		{"HAVING clause with allowed tables", "SELECT category_id, COUNT(*) as item_count FROM config_items GROUP BY category_id HAVING COUNT(*) > 10", []string{"config_"}, true},
+		{"HAVING clause with allowed tables", "SELECT category_id, COUNT(*) as item_count FROM config_items GROUP BY category_id HAVING COUNT(*) > 10", []string{"components_"}, false},
+
+		// Pivot
+		{"Pivot operation with allowed tables", "SELECT category_id, SUM(CASE WHEN value_type = 'A' THEN value ELSE 0 END) as A_value, SUM(CASE WHEN value_type = 'B' THEN value ELSE 0 END) as B_value FROM config_items GROUP BY category_id", []string{"config_"}, true},
+		{"Pivot operation with disallowed tables", "SELECT category_id, SUM(CASE WHEN value_type = 'A' THEN value ELSE 0 END) as A_value, SUM(CASE WHEN value_type = 'B' THEN value ELSE 0 END) as B_value FROM nonconfig_items GROUP BY category_id", []string{"config_"}, false},
+
+		// Conditional
+		{"Conditional expression with allowed tables", "SELECT id, name, value, CASE WHEN value >= 100 THEN 'High' ELSE 'Low' END as value_category FROM config_items", []string{"config_"}, true},
+		{"Conditional expression with disallowed tables", "SELECT id, name, value, CASE WHEN value >= 100 THEN 'High' ELSE 'Low' END as value_category FROM config_items", []string{"components_"}, false},
+
+		// Complex expressions
+		{"Complex expressions in SELECT with allowed tables", "SELECT id, CONCAT(name, ' - ', category_id) as full_name, value * 1.1 as adjusted_value FROM config_items", []string{"config_"}, true},
+		{"Complex expressions in SELECT with disallowed tables", "SELECT id, CONCAT(name, ' - ', category_id) as full_name, value * 1.1 as adjusted_value FROM config_items", []string{"components_"}, false},
+
+		// Aliases
+		{"Alias restricted table to allowed table", "SELECT config_users.* FROM (SELECT * FROM users) AS config_users", []string{"config_"}, false},
+		{"Alias allowed table to allowed table", "SELECT configs.* FROM (SELECT * FROM config_items) AS configs", []string{"config_"}, true},
+	}
+
+	for _, test := range tests {
+		output, err := validateTablesInQuery(test.query, test.allowedPrefix...)
+		if err != nil {
+			t.Logf("%s: validateTablesInQuery() returned an unexpected error: %v", test.name, err)
+			t.Fail()
+		}
+		if output != test.expectedOutput {
+			t.Logf("%s: validateTablesInQuery() returned %v, but expected %v", test.name, output, test.expectedOutput)
+			t.Fail()
+		}
+	}
+}
+
+// var _ = ginkgo.Describe("Query should only run non mutation queries", func() {
```

Comment:
This ginkgo spec is not moved to pkg_config_test.go because we're testing an unexported function.